src/Http/Response.php

```diff
@@ -655,7 +656,7 @@ public function getType()
      * @param string $contentType Either a file extension which will be mapped to a mime-type or a concrete mime-type.
      * @return static
      */
-    public function withType($contentType)
+    public function withType(string $contentType): self
```

Comment:
I couldn't use the return type as `ResponseInterface` here as it cause [this](https://travis-ci.org/cakephp/cakephp/jobs/409132848#L881).

Makes me think using return type as `ResponseInterface` for the `with*` methods is incorrect and all should be changed to `self`.